docs/rfcs/bun-app.ts

```diff
@@ -0,0 +1,240 @@
+  default: (req: Request, context: HandlerContext) => Promise<Response | null>;
+  // optional function that returns a list of imports
+  // these modules are loaded synchronously by Bun
+  // and passed into handler as context.imports
+  getImports?: (context: HandlerContext) => Import[];
+}
+
+type Import = { names: { [k: string]: string }; from: string };
+
+// the data that is passed as context to the Request handler
+// - manifest
+// - match: MatchedResult, only provided if `match` is specified in the `AppConfig`
+// - imports: only provided if `getImports` is specified in the `Handler`
+
+interface HandlerContext {
+  manifest?: BuildManifest;
+  match?: MatchedRoute;
+  imports: unknown; // depends on result of `getImports`
+}
+
+/////////////////////////////////////
+/////////////////////////////////////
+/////////     EXAMPLES     //////////
+/////////////////////////////////////
+/////////////////////////////////////
+
+// simple static file server
+{
+  const server = new App([
+    {
+      name: "static-server",
+      outdir: "./.build/client",
+      serve: [
+        {
+          mode: "static",
+          dir: "./public",
+          "style": "static",
+        },
+      ],
+    },
+  ]);
+
+  // serves files from `./public` on port 3000
+  await server.serve({
+    port: 3000,
+  });
+
+  // copies files from ./public to `.build/client`
+  await server.build();
+}
+
+// simple HTTP server
+{
+  /////////////////
+  // handler.tsx //
+  /////////////////
+  // @ts-ignore
+  export default (req: Request, ctx: BuildContext) => {
+    return new Response("hello world");
+  };
+
+  /////////////
+  // app.tsx //
+  /////////////
+  const app = new App([
+    {
+      name: "simple-http",
+      target: "bun",
+      entrypoints: [],
+      outdir: "./.build/server",
+      serve: [
+        {
+          mode: "handler",
+          handler: "./handler.tsx", // automatically included as entrypoing
+          prefix: "/api",
+        },
+      ],
+      // bundler config..
+    },
+  ]);
+
+  app.serve({
+    port: 3000,
+  });
+}
+
+// SSR react, pages directory
+{
+  /////////////////
+  // handler.tsx //
+  /////////////////
+  // @ts-ignore
+  import { renderToReadableStream } from "react-dom/server";
+
+  // @ts-ignore
+  export default (req: Request, context: HandlerContext) => {
+    const { manifest } = context;
+    const path = new URL(req.url).pathname;
+    const builtComponent = manifest.inputs.get(".pages/" + path + ".tsx");
+
+    const { default: Page } = await import(builtComponent);
+    const stream = renderToReadableStream(builtComponent);
+    return new Response(stream);
+  };
+
+  /////////////
+  // app.tsx //
+  /////////////
+  const router = new Bun.FileSystemRouter({
+    dir: "./app",
+    style: "nextjs",
+  });
+
+  const app = new App([
+    {
+      name: "react-ssr",
+      target: "bun",
+      entrypoints: ["./handler.tsx", ...Object.values(router.routes)], // path to static directory
```

Comment:
This breaks reloading/watching and is a pattern we should actively discourage

Instead, `entrypoints` needs to accept an `Array<string | FileSystemRouter>`.

Reply:
This is solved more fundamentally with the new `configs/routers` API. The "router" is defined upfront and references a bundler config by `label`.